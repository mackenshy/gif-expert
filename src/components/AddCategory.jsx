import PropTypes from "prop-types";
import { useState } from "react"

export function AddCategory({ onNewCategory }) {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const newInputValue = inputValue.trim();
        if(newInputValue.length <= 1 ) return;

        onNewCategory(newInputValue);
        setInputValue("");
    }

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input 
                type="text"
                placeholder="Search gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}