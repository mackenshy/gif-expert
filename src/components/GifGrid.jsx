import PropTypes from 'prop-types'
import { GifGridItem } from "./"
import { useFetchGifs } from "../hooks/useFetchGifs"

export function GifGrid({ category }) {
    const {images, isLoading} = useFetchGifs(category);

    return (
        <>
            <h3>{ category }</h3>
            { isLoading && (<h2>Loading...</h2>)}
            <div className="card-grid">
                {
                    images.map( gif => (
                        <GifGridItem 
                            key={gif.id} 
                            { ...gif }
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
