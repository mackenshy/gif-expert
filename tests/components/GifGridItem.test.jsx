import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../src/components/GifGridItem"

describe('GifGriditem', () => {
    const title = 'Saitama'
    const url = 'https://one-punch.com/saitam.gif'

    test('should match with snapshot', () => {
        const { container } = render(<GifGridItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })

    test('should show image with correct props', () => {
        render(<GifGridItem title={title} url={url} />)
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('should show title', () => {
        render(<GifGridItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy()
    })
})
