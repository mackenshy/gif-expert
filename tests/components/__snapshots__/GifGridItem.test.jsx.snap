// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGriditem should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitam.gif"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
